### DISCUSSION TITLE
Does the push/pull workflow support Arnold shaders?

### DISCUSSION THREAD

**Discussion by sharktacos** [Autodesk/maya-usd#2741] [Q&A]
When pulling a staged USD file with Arnold shaders into Maya (Edit as Maya Data) the shader info is lost and a grey default Lambert1 material is assigned. 

Similarly, if an Arnold shader is then assigned and the geo is pushed back to USD (Merge Maya Edits to USD) with the plugin configuration set to "Arnold" nevertheless none of the shader info transfers over and instead a white shader is assigned to the USD. Is this a known issue? 

This is in Maya 2023, MayaUSDPlugin 0.16.0

**Comment by cdwasden** [comment]
Hi sharktacos, 

I think we were running into a similar issue with our initial USD tests. We were testing with version 0.18, I believe. With that version, the Arnold materials were exported with the USD file - but they were saved in a separate place to other materials in the file. Other materials went into the "Looks" scope and the Arnold materials were saved outside of that scope. I believe we could read that single file as a USD stage and the materials did come back into the outliner, but if we tried to add the USD file as a reference, the materials outside of the Looks scope would not be imported as well and we would get the white shader you were talking about.

I ended up writing a script that would manipulate the USD file and move the Arnold materials within the "Looks" scope and repath all references to the Arnold materials so they would be correctly found. After that, we could import the USD file and the Arnold materials would come along as well.

I don't have much experience with 0.16, so your issue might be different, but can you see that the Arnold materials are actually present within your USD file? In our experience we were able to export the materials and manipulate them afterwards. Moving them into the "scope" might help repoint them so Maya and Arnold can correctly find them again.

**Reply by sharktacos** [reply]
@cdwasden Thanks for your reply!  Should this be logged as an issue then?
